src/ape/exceptions.py

```diff
@@ -832,3 +832,9 @@ def _get_custom_python_traceback(
         tb_next = tb
 
     return frames[-1]
+
+
+class SendingFundsToNonPayableConstructorError(ApeException):
```

Comment:
I think renaming to this and put it before `TransactionError`
```suggestion
class MethodNonPayableError(ContractError):
```